vsintegration/tests/UnitTests/TestLib.LanguageService.fs

```diff
@@ -261,7 +261,48 @@ type LanguageServiceBaseTests() =
     // Timings ----------------------------------------------------------------------------- 
     let stopWatch = new Stopwatch()
     let ResetStopWatch() = stopWatch.Reset(); stopWatch.Start()
-               
+      
+
+    let Init() =
+        let AssertNotAssemblyNameContains(a:System.Reflection.Assembly, text1:string, text2:string) = 
+            let fullname = sprintf "%A" a
+            if fullname.Contains(text1) && fullname.Contains(text2) then
+                // Can't throw an exception here because its in an event handler.
+                System.Diagnostics.Debug.Assert(false, sprintf "Unexpected: loaded assembly '%s' to not contain '%s' and '%s'" fullname text1 text2)
+
+        // Under .NET 4.0 we don't allow 3.5.0.0 assemblies
+        let AssertNotBackVersionAssembly(args:AssemblyLoadEventArgs) =
+
+            // We're worried about loading these when running against .NET 4.0:
+            // Microsoft.Build.Tasks.v3.5, Version=3.5.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
+            // Microsoft.Build.Utilities.v3.5, Version=3.5.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
+            AssertNotAssemblyNameContains(args.LoadedAssembly,"Microsoft.Build", "Version=3.5.0.0") 
+            ()
+        AppDomain.CurrentDomain.AssemblyLoad.Add AssertNotBackVersionAssembly
+
+        UIStuff.SetupSynchronizationContext()
+
+        defaultVS <- ops.CreateVisualStudio()
+        currentVS <- defaultVS
+        
+        defaultSolution <- GlobalFunctions.CreateSolution(defaultVS)
+        cache.Clear()
+
+    do
+        Init()
```

Comment:
This mixes what used to be a onetimesetup (per namespace?) execution and per test clas execution.
Why?

(a `do` in a base type means code going to the constructor, which is then called from every subtype)

Reply:
Basically because it's the easiest way to do that and it doesn't have any effect on performance.

Xunit doesn't really encourage doing complex setups for tests - it's possible but we'd need to create an extra fixture and propagate it in many places, or refactor the tests. Given that we don't want to invest in this suite, I went ahead with the simple way.